tails_files/install.sh

```diff
@@ -162,13 +162,20 @@ if [ -f $XSESSION_RC ]; then
     fi
 fi
 
+# remove previous NetworkManager hook if present. it runs later than we'd like
+OLD_NM_HOOK=$TAILSCFG/custom-nm-hooks/99-tor-reload.sh
+if [ -f $OLD_NM_HOOK ]; then
+    rm -f $OLD_NM_HOOK > /dev/null 2>&1
+    rm -f $INSTALL_DIR/99-tor-reload.sh > /dev/null 2>&1
+fi
```

Comment:
Under no circumstances should the file `/etc/NetworkManager/dispatcher.d/99-tor-reload.sh` exist. In lieu of Ansible's `state=absent` pragma for ensuring file absence, the best we have in a bash script is `rm -f <path>`.

Right now the script checks if `/home/amnesia/Persistent/.securedrop/99-tor-reload.sh` exists, and removes `/etc/NetworkManager/dispatcher.d/99-tor-reload.sh` **if-and-only-if** it does. I see the presumed association here, but the conditional logic is actually implicit, not explicit, since it assumes that the existence of a file is predicated on the existence of another.

It's an edge case, but if I ctrl+c this script at the right moment, I can create a system state where two redundant nm hooks are present. That's not good, and I definitely don't want to debug that state months from now.

I suggest refactoring the conditional block as:

```
# Remove previous NetworkManager hook if present. The "99-" prefix
# caused the hook to run too late for apt persistence to work reliably.
for d in $TAILSCFG $INSTALL_DIR /etc/NetworkManager/dispatcher.d ; do
    rm -f "$d/99-tor-reload.sh" > /dev/null 2>&1
done
```

It may be more explicit to check for file existence per file within the for-loop—personally I find that a bit obtuse, but it's mostly a stylistic question at that point, and the per-file checks would still ensure file absence as intended.